projects/observability/src/shared/components/cartesian/d3/legend/cartesian-legend.ts

```diff
@@ -23,7 +35,14 @@ export class CartesianLegend {
     private readonly injector: Injector,
     private readonly intervalData?: CartesianIntervalData,
     private readonly summaries: Summary[] = []
-  ) {}
+  ) {
+    this.activeSeries = [...this.series];
+    this.initialSeries = [...this.series];
+    this.activeSeries$ = this.activeSeriesSubject.asObservable().pipe(
+      startWith(undefined),
+      switchMap(() => of(this.activeSeries))
```

Comment:
a. this need not use switchMap and wrap the series, it can just use `mapTo`
b. Why not make the subject hold the series rather than having it as a notification and then mapping it? Every place we `next` it we have the new value, so it's just an extra layer of code
c. We should make sure these are distinct - either with an operator like distinctUntilChanged (which would need to use isEqual due to the changing references like `distinctUntilChanged(isEqual)`), or by only invoking it on change. The only spot that I can see where we're potentially doubling up is when changing modes. If there's a single series (commented on preventing this elsewhere), selecting it won't change the activeSeries selection. Similarly, if all series are selected and we click reset, only the mode changes, not the activeSeries.